src/gps/GPS.cpp

```diff
@@ -625,659 +630,678 @@ void GPS::publishUpdate()
+        return false;
+    }
 
-bool GPS::whileIdle()
-{
-    int charsInBuf = 0;
-    bool isValid = false;
-    if (!isAwake) {
-        clearBuffer();
-        return isAwake;
+    bool GPS::hasFlow()
+    {
+        return reader.passedChecksum() > 0;
     }
+
+    bool GPS::whileIdle()
+    {
+        int charsInBuf = 0;
+        bool isValid = false;
+        if (!isAwake) {
+            clearBuffer();
+            return isAwake;
+        }
 #ifdef SERIAL_BUFFER_SIZE
-    if (_serial_gps->available() >= SERIAL_BUFFER_SIZE - 1) {
-        LOG_WARN("GPS Buffer full with %u bytes waiting. Flushing to avoid corruption.\n", _serial_gps->available());
-        clearBuffer();
-    }
+        if (_serial_gps->available() >= SERIAL_BUFFER_SIZE - 1) {
+            LOG_WARN("GPS Buffer full with %u bytes waiting. Flushing to avoid corruption.\n", _serial_gps->available());
+            clearBuffer();
+        }
 #endif
-    // if (_serial_gps->available() > 0)
-    // LOG_DEBUG("GPS Bytes Waiting: %u\n", _serial_gps->available());
-    // First consume any chars that have piled up at the receiver
-    while (_serial_gps->available() > 0) {
-        int c = _serial_gps->read();
-        UBXscratch[charsInBuf] = c;
+        // if (_serial_gps->available() > 0)
+        // LOG_DEBUG("GPS Bytes Waiting: %u\n", _serial_gps->available());
+        // First consume any chars that have piled up at the receiver
+        while (_serial_gps->available() > 0) {
+            int c = _serial_gps->read();
+            UBXscratch[charsInBuf] = c;
 #ifdef GPS_DEBUG
-        LOG_DEBUG("%c", c);
+            LOG_DEBUG("%c", c);
 #endif
-        isValid |= reader.encode(c);
-        if (charsInBuf > sizeof(UBXscratch) - 10 || c == '\r') {
-            if (strnstr((char *)UBXscratch, "$GPTXT,01,01,02,u-blox ag - www.u-blox.com*50", charsInBuf)) {
-                rebootsSeen++;
+            isValid |= reader.encode(c);
+            if (charsInBuf > sizeof(UBXscratch) - 10 || c == '\r') {
+                if (strnstr((char *)UBXscratch, "$GPTXT,01,01,02,u-blox ag - www.u-blox.com*50", charsInBuf)) {
+                    rebootsSeen++;
+                }
+                charsInBuf = 0;
+            } else {
+                charsInBuf++;
             }
-            charsInBuf = 0;
-        } else {
-            charsInBuf++;
         }
+        return isValid;
+    }
+    void GPS::enable()
+    {
+        enabled = true;
+        setInterval(GPS_THREAD_INTERVAL);
+        setAwake(true);
     }
-    return isValid;
-}
-void GPS::enable()
-{
-    enabled = true;
-    setInterval(GPS_THREAD_INTERVAL);
-    setAwake(true);
-}
 
-int32_t GPS::disable()
-{
-    enabled = false;
-    setInterval(INT32_MAX);
-    setAwake(false);
+    int32_t GPS::disable()
+    {
+        enabled = false;
+        setInterval(INT32_MAX);
+        setAwake(false);
+
+        return INT32_MAX;
+    }
 
-    return INT32_MAX;
-}
\ No newline at end of file
+    void GPS::toggleGpsMode()
+    {
```

Comment:
Wouldn't it be simpler (and more logical) to just toggle between enabled/disabled and if not present it does nothing?